smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/AddCompressionDependency.java

```diff
@@ -0,0 +1,149 @@
+                        writer.addImport("loadConfig", "loadNodeConfig",
+                                TypeScriptDependency.NODE_CONFIG_PROVIDER);
+                        writer.addImport("NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_OPTIONS", null,
+                                TypeScriptDependency.MIDDLEWARE_COMPRESSION);
+                        writer.write("loadNodeConfig(NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_OPTIONS)");
+                    }
+                );
+            case BROWSER:
+                return MapUtils.of(
+                    "disableRequestCompression", writer -> {
+                        writer.addDependency(TypeScriptDependency.MIDDLEWARE_COMPRESSION);
+                        writer.addImport("DEFAULT_DISABLE_REQUEST_COMPRESSION", null,
+                            TypeScriptDependency.MIDDLEWARE_COMPRESSION);
+                        writer.write("DEFAULT_DISABLE_REQUEST_COMPRESSION");
+                    },
+                    "requestMinCompressionSizeBytes", writer -> {
+                        writer.addDependency(TypeScriptDependency.MIDDLEWARE_COMPRESSION);
+                        writer.addImport("DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES", null,
+                            TypeScriptDependency.MIDDLEWARE_COMPRESSION);
+                        writer.write("DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES");
+                    }
+                );
+            default:
+                return Collections.emptyMap();
+        }
+    }
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return ListUtils.of(
+            RuntimeClientPlugin.builder()
+                .withConventions(TypeScriptDependency.MIDDLEWARE_COMPRESSION.dependency,
+                    "Compression", RuntimeClientPlugin.Convention.HAS_CONFIG)
+                .servicePredicate((m, s) -> hasRequestCompressionTrait(m, s))
+                .build(),
+            RuntimeClientPlugin.builder()
+                .withConventions(TypeScriptDependency.MIDDLEWARE_COMPRESSION.dependency,
+                    "Compression", RuntimeClientPlugin.Convention.HAS_MIDDLEWARE)
+                .additionalPluginFunctionParamsSupplier((m, s, o) -> getPluginFunctionParams(m, s, o))
+                .operationPredicate((m, s, o) -> hasRequestCompressionTrait(o))
+                .build()
+        );
+    }
+
+    private static Map<String, Object> getPluginFunctionParams(
+        Model model,
+        ServiceShape service,
+        OperationShape operation
+    ) {
+        Map<String, Object> params = new TreeMap<String, Object>();
+
+        // Populate encodings from requestCompression trait
+        RequestCompressionTrait requestCompressionTrait = operation.expectTrait(RequestCompressionTrait.class);
+        params.put("encodings", requestCompressionTrait.getEncodings());
+
+        return params;
+    }
+
+    // return true if operation shape is decorated with `httpChecksum` trait.
```

Comment:
this should not say httpChecksum, right?

Reply:
```suggestion
    // return true if operation shape is decorated with `requestCompression` trait.
```